base/src/main/java/org/mozilla/jss/provider/javax/crypto/JSSTrustManager.java

```diff
@@ -179,6 +191,9 @@ public void checkCertChain(X509Certificate[] certChain, String keyUsage) throws
 
         Enumeration<ValidityItem> reasons = status.getReasons();
         if (!reasons.hasMoreElements()) {
+            if (enabledNSSCertVerify) {
+                nssCertVerify(certChain, keyUsage);
```

Comment:
In line 201 there's a code that will allow the user to manually accept the cert even though it's considered invalid. Should cert revocation be something that can be manually accepted as well? If so, the `nssCertVerify()` probably should be called before line 193.

Reply:
This is problematic. Current JSS certificate verification with NSS throw an exception if the verification fails but the error code is reported only in the message and I do not like the message parsing approach to get the problem. 

We can implement a new JSS function to get the code instead of the exception but  NSS verification report the first problem with the certificate so it is possible to fail the verification before it checks if they are revoked. Not sure if this can work.

Reply:
Yeah, internally JSS should implement native methods that can fully control NSS so we can get everything that NSS provides, but externally JSS should provide standard Java security API so eventually it can be used without JSS-specific code.